src/app/editor/contextView.js

```diff
@@ -129,11 +130,54 @@ class ContextView {
       }
     }
 
+    function next () {
+      var nodes = self._api.contextualListener.getActiveHighlights()
+      var currentName = node.attributes.name
+      var position = self.sourceMappingDecoder.decode(node.src)
+
+      if (currentName !== self.refName) self.ref = 0
+      var k = self.ref
+
+      /// LOGS ///
+      nodes.forEach(node => console.log(node.position.start))
+      console.log('----')
+      console.log(k)
+      console.log('----')
+
+      var pos = nodes[k].position
+      self._api.jumpTo(pos)
+
+      self.ref = (self.ref + 1) % nodes.length
+      self.refName = currentName
+    }
+
+    function previous () {
+      var nodes = self._api.contextualListener.getActiveHighlights()
+      var currentName = node.attributes.name
+      if (currentName !== self.refName) self.ref = nodes.length - 1
```

Comment:
I think you really don't need `self.refName` . is reseting the counter to 0 when `_render` is called works?

Reply:
@yann300 I want to reset it only when new keyword is selected. 

When I click one selector and then click through all the references, I want this.ref to increase (this.ref = (this.ref + 1) % nodes.length), but when I select a new selector, I want to reset it.

_render seem to run with each click on next and previous, so I can't reset it each time. Or am I missing something?